awscli/customizations/ecs/deploy.py

```diff
@@ -122,7 +126,8 @@ def _run_main(self, parsed_args, parsed_globals):
             region_name=parsed_globals.region,
             verify=parsed_globals.verify_ssl)
 
-        self._validate_code_deploy_resources(codedeploy_client)
+        self.wait_time = \
+            self._validate_code_deploy_resources(codedeploy_client)
```

Comment:
It seems strange that the validator would return the wait time. I think I would prefer a separate call to get the wait time after validation has occured.

Reply:
Do you mean a separate API call or a separate private method in `_run_main` that retrieves the wait_time? Since the `CodeDeployValidator` retrieves all the info we need to determine the wait time, I'd rather do that latter, perhaps by keeping the validator around as an attribute of the main `ECSDeploy` obj so we can use it later. I'll add that change so you can LMK if it's enough of an improvement.